packages/data/addon/services/bard-metadata.js

```diff
@@ -59,9 +65,9 @@ export default Service.extend({
    * @param {Object} options - options object used by the adapter
    * @returns {Promise} promise that loads metadata
    */
-  loadMetadata(options) {
+  loadMetadata(options = {}) {
     //fetch metadata from WS if metadata not yet loaded
-    if (!get(this, 'metadataLoaded')) {
+    if (!get(this, 'metadataLoaded') || !this.loadedDataSources.includes(options.dataSourceName)) {
```

Comment:
I find this a little easier to read, but I'm not sure how useful `metadataLoaded` is now
```suggestion
    if (!(get(this, 'metadataLoaded') && this.loadedDataSources.includes(options.dataSourceName))) {
```

Reply:
Yeah we can probably rid of it in another PR.